examples/mnist_test_oss.py

```diff
@@ -0,0 +1,169 @@
+## adapted from https://github.com/pytorch/examples/blob/master/mnist/main.py
+from __future__ import print_function
+import argparse
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+
+import time
+from fairscale.optim.oss import OSS
+import torch.multiprocessing as mp
+from fairscale.nn.data_parallel import ShardedDataParallel
+
+# how is world_size defined?
+WORLD_SIZE = 2
+OPTIM = torch.optim.RMSprop
+BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
+
+def dist_init(rank, world_size, backend):
+    print(f"Using backend: {backend}")
+    dist.init_process_group(backend=backend, init_method="tcp://localhost:29501", rank=rank, world_size=world_size)
+
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 32, 3, 1)
+        self.conv2 = nn.Conv2d(32, 64, 3, 1)
+        self.dropout1 = nn.Dropout2d(0.25)
+        self.dropout2 = nn.Dropout2d(0.5)
+        self.fc1 = nn.Linear(9216, 128)
+        self.fc2 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = self.conv2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.dropout1(x)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.dropout2(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+
+
+def train(rank, args, model, device, train_loader, num_epochs):
+    ##############
+    # SETUP 
+    dist_init(rank, WORLD_SIZE, BACKEND)
+    ddp = ShardedDataParallel(
+        module=model, optimizer=torch.optim.Adadelta, optimizer_params={"lr": 1e-4}, world_size=WORLD_SIZE
+    )
+    optimizer = ddp.optimizer
+    # Reset the memory use counter
+    torch.cuda.reset_peak_memory_stats(rank)
+
+    # Dummy training loop
```

Comment:
dummy <> standard  ? In the other tuto I wrote "dummy" to try to say that the changes were not there (and it could well be a poor choice of words), but here if you want a working example then maybe that dummy does not convey the right meaning ?